cmake/FindCVC4.cmake

```diff
@@ -1,4 +1,8 @@
-find_path(CVC4_INCLUDE_DIR cvc4/cvc4.h)
-find_library(CVC4_LIBRARY NAMES cvc4 )
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(CVC4 DEFAULT_MSG CVC4_LIBRARY CVC4_INCLUDE_DIR)
+if (USE_CVC4)
+    find_path(CVC4_INCLUDE_DIR cvc4/cvc4.h)
+    find_library(CVC4_LIBRARY NAMES cvc4 )
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(CVC4 DEFAULT_MSG CVC4_LIBRARY CVC4_INCLUDE_DIR)
+else()
+    set(CVC4_FOUND "${CVC4_FOUND}" "false")
```

Comment:
Why the ``"${CVC4_FOUND}"`` here?

Reply:
As I explained above, in the comments, the results of find_package sets a variable named PAKCAGE_FOUND. In this case, CVC4_FOUND. This tricks the setup into thinking it was not found, without the need to modify the logic of CMakeLists.cmake from libsolidity.

A message was added to provide additional info for this case.

Reply:
This code in each ``FindPackage.cmake`` could also go inside ``libsolidity/CMakeLists.cmake`` after each find is executed.

Reply:
I'm aware of CMake's ``*_FOUND`` variables and I understand what the intention here is (sorry for the confusion), but I'm wondering why not just use ``set(CVC4_FOUND "false")`` here?
To me it looks like ``set(CVC4_FOUND "${CVC4_FOUND}" "false")`` would result in ``true;false``, if ``CVC4_FOUND`` was ``true`` beforehand - which is not what we want. Probably it will never be set before, but then we can just remove ``"${CVC4_FOUND}"`` from this ``set`` command and use ``set(CVC4_FOUND "false")``, or am I missing something?

Reply:
You're not missing anything, you're right :)